git_theta/lsh/pool.py

```diff
@@ -0,0 +1,64 @@
+"""Class for deterministically supplying pre-computed random values"""
+
+import numpy as np
+import numba as nb
+from numba import int32, float32
+import sys
+
+# TODO(bdlester): importlib.resources doesn't have the `.files` API until python
+# version `3.9` so use the backport even if using a python version that has
+# `importlib.resources`.
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+
+# RandomnessPool is a singleton class so we don't read pool data off disk multiple times per git-theta command
+SINGLETON = None
+
+spec = [("pool", nb.float64[:]), ("index_hashes", nb.int64[:])]
+
+
+@nb.experimental.jitclass(spec)
+class RandomnessPool:
+    def __init__(self, pool, index_hashes):
+        self.pool = pool
+        self.index_hashes = index_hashes
+
+    def get_hyperplanes(self, feature_size):
```

Comment:
I don't think you need two functions here, something like

```python
def get_hyperplanes(self, input_features, signature_idx):
  return self.pool[
    np.mod(np.bitwise_xor(input_features, self.index_hashes[signature_idx]), self.pool.size, dtype=np.int64)
  ] 
```

Then you can get all the hyperplanes at once with

```
pool.get_hyperplanes(
  np.expand_dims(np.arange(0, feature_size, dtype=np.int64)),
  np.arange(0, self.index_hashes.size, dtype=np.int64)
)
```

and a single element by just passing in the feature idx and the signature idx

Reply:
I tried this out and it doesn't seem to work because numba doesn't support all types of advanced indexing, which is what's going on in `self.pool[...]` (https://numba.pydata.org/numba-doc/dev/reference/numpysupported.html)